src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
+    public T [] grow() {
+        T[] sourceArray = (T[]) values;
+        values = new Object[size + (size >> 1)];
+        System.arraycopy(sourceArray, 0, values, 0, size);
+        return sourceArray;
+    }
+
+    public void chekIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index:" + index + "out of size" + size);
+        }
+    }
+
+    public void checkIndexForAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index:" + index + "out of size" + size);
+        }
+    }
```

Comment:
those should be private methods and should be in the end ( after public methods )